build.gradle

```diff
@@ -12,10 +12,10 @@ repositories {
 }
 
 dependencies {
-    runtime 'com.sun.faces:jsf-api:2.1.0'
-    runtime 'javax.servlet:javax.servlet-api:3.0.1'
-    runtime 'log4j:log4j:1.2.16'
-    runtime group: 'org.springframework', name: 'spring-context', version: '[3.+,4.+]'
+    compile 'com.sun.faces:jsf-api:2.1.0'
+    compile 'javax.servlet:javax.servlet-api:3.0.1'
+    compile 'org.apache.logging.log4j:log4j-api:2.7'
+    compile 'org.springframework:spring-context:4.3.5.RELEASE'
```

Comment:
The reason I was doing this PR was that I was forced by this library to use Spring3.  Therefore defined it only as compile time dependency which at least doesn't bring in all dependencies with. But yes, it works in this context only with Spring 4 (any version) and in the initial one only with Spring3 but even worse because one has suddenly two different Spring versions on the Classloader which is terrible.

So my solution is just a workaround. The best solution would be to be indpendent from Spring at all with any library. To be dependent on Spring is never a good idea because of the mentioned problems above.

Thererfore I sitll propose to have the dependencies as compile time only and not delivering the dependencies with the library.

With my change, it works perfect in any Spring 4 environement.

Reply:
BTW, I even changed it to compileOnly in my fork, works well. It is basically the same as the maven "provided" which is in my understanding here the right thing.
Let me know if you are interested in a new PR.

Reply:
>  I was forced by this library to use Spring3. 
I use spring 4 with with library without any issues. Spring 3 is minimal dependency, and you must use spring here because it is actually meant tow work with spring.

> to have the dependencies as compile time
Compile time dependency force you to use this dependency, runtime does not.

AFAIK runtime in gradle is a workaround for maven provided scope.

Reply:
Don't agree really, my knowledge is that compile, and in newer version compileOnly are the same as maven provided. And my experience proofed it. I had to exclude all Spring3 stuff from my maven profile in order to get rid of Spring3 dependency. It didn't work anyhow. Therefore I forked and made it dependent on Spring4. And by using compileOnly it doesn't bring any dependencies to the runtime and takes whatever it finds in the classpath.

At least I interpreted the following article like this: https://blog.gradle.org/introducing-compile-only-dependencies